src/io/flutter/run/FlutterRunner.java

```diff
@@ -52,15 +52,17 @@ public boolean canRun(final @NotNull String executorId, final @NotNull RunProfil
       return false;
     }
 
-    final FlutterRunConfiguration runConfiguration = (FlutterRunConfiguration)profile;
+    final FlutterRunConfigurationBase runConfiguration = (FlutterRunConfigurationBase)profile;
     final Project project = runConfiguration.getProject();
-    final FlutterSettings settings = FlutterSettings.getInstance(project);
 
-    if (settings.isShowDevices()) {
-      final FlutterDaemonService service = FlutterDaemonService.getInstance(project);
-      if (service == null || !service.hasSelectedDevice()) {
-        return false;
-      }
+    if (FlutterSdk.getFlutterSdk(project) == null) {
+      return false;
+    }
+
+    final FlutterDaemonService service = FlutterDaemonService.getInstance(project);
+    //noinspection SimplifiableIfStatement
+    if (!service.isActive() || !service.hasSelectedDevice()) {
```

Comment:
I think we need a null check here?:

    if (service == null || !service.isActive() || !service.hasSelectedDevice())